### PR TITLE
Ex3

### PR DESCRIPTION
**Pull request by stevelee2497**

### CODE REVIEW

File: Exercise3/Models/Ultilities.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+                    mList.Add(new ManagerItem(Resource.Drawable.file32, name));
+                }
+            }
+            return mList;
+        }
+
+        private static bool IsImage(string name)
+        {
+            var imageTag = new[] { ".png", ".jpg", ".gif" };
+            return imageTag.Any(name.Contains);
```
Comment (voquanghoa) [inline Exercise3/Models/Ultilities.cs]: Need to change
- Replace `Contains` to `EndWith`
- if the file name is abc.PNG, this function will not work well.

File: Exercise3/Models/Ultilities.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+                {
+                    mList.Add(new ManagerItem(Resource.Drawable.file32, name));
+                }
+            }
+            return mList;
+        }
+
+        private static bool IsImage(string name)
+        {
+            var imageTag = new[] { ".png", ".jpg", ".gif" };
```
Comment (voquanghoa) [inline Exercise3/Models/Ultilities.cs]: Rename `imageTag` to `imageExtension`

File: Exercise3/Models/ManagerItem.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
+﻿using System.Runtime.Serialization;
+
+namespace Exercise3.Models
+{
+    public class ManagerItem : ISerializable
```
Comment (voquanghoa) [inline Exercise3/Models/ManagerItem.cs]: Why you need to implement `ISerializable`

File: Exercise3/Models/ManagerItem.cs [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+        public ManagerItem(int image, string name)
+        {
+            Image = image;
+            Name = name;
+        }
+
+        public void GetObjectData(SerializationInfo info, StreamingContext context)
+        {
+            throw new System.NotImplementedException();
```
Comment (voquanghoa) [inline Exercise3/Models/ManagerItem.cs]: wtf

File: Exercise3/Models/Ultilities.cs [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+using System.Linq;
+
+namespace Exercise3.Models
+{
+    public class Ultilities
+    {
+        public static List<ManagerItem> GetFileAndFolderNames(string path)
+        {
+            var mList = new List<ManagerItem>();
+            var fileAndFolderNames = Directory.GetFileSystemEntries(path);
```
Comment (voquanghoa) [inline Exercise3/Models/Ultilities.cs]: On Linux system (which android actually extended), everything is files, including files, folders, drives. So the name here is `fileNames`

File: Exercise3/Models/Ultilities.cs [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+        private static bool IsImage(string name)
+        {
+            var imageTag = new[] { ".png", ".jpg", ".gif" };
+            return imageTag.Any(name.Contains);
+        }
+
+        private static bool IsMediaFile(string name)
+        {
+            var mediaTag = new[] { ".avi", ".mp4", ".mkv", ".flv", ".mp3" };
```
Comment (voquanghoa) [inline Exercise3/Models/Ultilities.cs]: The same thing as before

File: Exercise3/Views/MainActivity.cs [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+        {
+            var intent = new Intent(this, typeof(OpenFile));
+            intent.PutExtra(FileType, managerItem.Image);
+            intent.PutExtra(FileName, managerItem.Name);
+            StartActivity(intent);
+        }
+
+        public override void OnBackPressed()
+        {
+            if (path == Environment.ExternalStorageDirectory.AbsolutePath)
```
Comment (voquanghoa) [inline Exercise3/Views/MainActivity.cs]: use `string.equals(...)` instead please.

File: Exercise3/Models/ManagerItem.cs (hunk 3 above)
Comment (stevelee2497) [reply Exercise3/Models/ManagerItem.cs]: My bad, I forgot to delete it.
Done!

File: Exercise3/Models/ManagerItem.cs (hunk 4 above)
Comment (stevelee2497) [reply Exercise3/Models/ManagerItem.cs]: it's auto implemented

File: Exercise3/Models/Ultilities.cs (hunk 5 above)
Comment (stevelee2497) [reply Exercise3/Models/Ultilities.cs]: done

File: Exercise3/Models/Ultilities.cs (hunk 2 above)
Comment (stevelee2497) [reply Exercise3/Models/Ultilities.cs]: done

File: Exercise3/Models/Ultilities.cs (hunk 1 above)
Comment (stevelee2497) [reply Exercise3/Models/Ultilities.cs]: done

File: Exercise3/Models/Ultilities.cs (hunk 6 above)
Comment (stevelee2497) [reply Exercise3/Models/Ultilities.cs]: done

File: Exercise3/Views/MainActivity.cs (hunk 7 above)
Comment (stevelee2497) [reply Exercise3/Views/MainActivity.cs]: my bad

**Pull request merged**